public/script.js

```diff
@@ -17,3 +17,72 @@ if ("serviceWorker" in navigator) {
       console.error(err);
     });
 }
+
+function userAgentDetectDevice(){
+  // if the userAgent contains "Windows"
+  if(navigator.userAgent.indexOf("Windows") !== -1){
+    return "Windows"
+  }
+  // if the userAgent contains "Macintosh"
+  else if(navigator.userAgent.indexOf("Macintosh") !== -1){
+    return "macOS"
+  }
+  // if the userAgent contains "Linux"
+  else if(navigator.userAgent.indexOf("Linux") !== -1){
+    return "Linux"
+  }
+  // if the userAgent contains "Android"
+  else if(navigator.userAgent.indexOf("Android") !== -1){
+    return "Android"
+  }
+  // if the userAgent contains "iPhone" or "iPad" or "iPod"
+  else if(navigator.userAgent.indexOf("iPhone") !== -1 || navigator.userAgent.indexOf("iPad") !== -1 || navigator.userAgent.indexOf("iPod") !== -1){
+    return "iOS"
+  }
+}
```

Comment:
I think there's a bug here.

On line 80, you're checking `if(device === ""){`, but if none of the `if` statements match, then it's going to implicitly return `undefined` instead. You can resolve this wither by adding a `return "";` at the end here, or by changing the condition on line 80 to something like `if (!device) {` so any falsy value will be caught.

I think I prefer the `return "";` myself, but I'll leave it up to you.